File: Code/Disney/disney.xBandController/src/windows/Test/Disney.xBand.IDMS/Parties/PartyDao.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Disney.xBand.IDMS.Parties
{
    public class PartyDao : Dao.DaoBase, IPartyDao
    {
        public PartyDao(string rootUrl)
            : base(rootUrl)
        {
        }

        public Party Get(long partyId, Metrics metrics)
        {
            return GetRequest<Party>(String.Concat(this.RootUrl, String.Format("party/{0}",partyId)), metrics);
        }

        public Party Get(string partyName, Metrics metrics)
        {
            return GetRequest<Party>(String.Concat(this.RootUrl, String.Format("party/search/{0}", partyName)), metrics);
        }

        public long Save(Party party, Metrics metrics)
        {
            return PostRequest<Party>(String.Concat(this.RootUrl, "party/"), metrics, party);
        }

        public long Assign(GuestPartyAssign guestPartyAssign, Metrics metrics)
        {
            return PostRequest<GuestPartyAssign>(String.Concat(this.RootUrl, "party/guest"), metrics, guestPartyAssign);
        }
    }
}

File: Code/Disney/disney.xBandController/src/linux/IDMSLib/src/main/java/com/disney/xband/idms/dao/PartyDAO.java
package com.disney.xband.idms.dao;

import com.disney.xband.idms.lib.model.Party;

public interface PartyDAO 
{
	public Party getParty(String partyName) throws DAOException;

	public Party getParty(long partyId) throws DAOException;

	public Party getParty(String identifierType, 
			String identifierValue) throws DAOException;
	
	public long createParty(Party party) throws DAOException;
	
	public boolean addGuestToParty(long guestId, long partyId) 
			throws DAOException;
}
